### PR TITLE
Fixes RHICOMPL-37: Handle empty/StringIO uploads

### PR DESCRIPTION
**Pull request by akofink**
I'm not sure how this ever worked :)

### CODE REVIEW

**Comment by quarckster** [comment]
@akofink could please add tests for that?

File: app/services/safe_downloader.rb [hunk 1]
```diff
@@ -48,6 +48,7 @@ def encode_url(url)
     def patch_if_less_than_10k(downloaded_file, tempfile)
       return downloaded_file unless downloaded_file.is_a?(StringIO)
 
+      tempfile = Tempfile.new
```
Comment (quarckster) [inline app/services/safe_downloader.rb]: Don't you receive `tempfile` as an argument in this method, do you?

Comment (akofink) [reply app/services/safe_downloader.rb]: Wow, yeah - let me redo this PR.

**Comment by akofink** [comment]
This is fixed by https://github.com/RedHatInsights/compliance-backend/pull/50

**Pull request closed without merging**